### PR TITLE
Minor cleanups

### PR DESCRIPTION
**Pull request by moggers87**
## Description

- Remove references to Django 1.11
- Remove dead code
- Update example code in docs
- Test should print warnings
- Replace deprecated functions
  * Replace `url()` with `re_path()`
  * Replace assertEquals with assertEqual
- Resolve LOGOUT_URL warning (and add tests for it)
- Run flake8 on the same files as isort
  * And fix flake8 errors that weren't being reported before

## Motivation and Context

Improves coverage, future proofs code, and declutters tox.ini

## How Has This Been Tested?

CI passes

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by claudep** [comment]
Note you might also consider replacing the `url()` functions by `path()` in  `urls.py`.

**Comment by moggers87** [comment]
Did I miss anything?

File: example/urls.py [hunk 1]
```diff
@@ -11,39 +11,39 @@
...
         name='registration_complete',
     ),
-    url(r'', include(tf_urls)),
-    url(r'', include(tf_twilio_urls)),
-    url(r'', include('user_sessions.urls', 'user_sessions')),
-    url(r'^admin/', admin.site.urls),
+    re_path(r'', include(tf_urls)),
+    re_path(r'', include(tf_twilio_urls)),
+    re_path(r'', include('user_sessions.urls', 'user_sessions')),
+    re_path(r'^admin/', admin.site.urls),
```
Comment (claudep) [inline example/urls.py:42]: `re_path` should be limited to cases where a regular expression is unavoidable. In this file, I don't see any of these, so using `path` is IMHO better and simpler (e.g. `path('account/register/',…)`).

File: tests/test_views_login.py [hunk 2]
```diff
@@ -450,29 +450,46 @@ def test_generate(self):
                          response.context_data['device'].token_set.all()])
         self.assertNotEqual(first_set, second_set)
 
+    def test_no_cancel_url(self):
+        response = self.client.get(reverse('two_factor:login'))
+        self.assertEqual(response.status_code, 200)
+        self.assertNotIn('canel_url', response.context.keys())
```
Comment (claudep) [inline tests/test_views_login.py:456]: `cancel_url`?

File: tests/urls.py [hunk 3]
```diff
@@ -8,44 +8,44 @@
...
-                                verification_url='/account/login/'),
+    re_path(
+        r'^secure/redirect_unverified/$',
+        SecureView.as_view(raise_anonymous=True,
+                           verification_url='/account/login/'),
     ),
-    url(r'', include(tf_urls)),
-    url(r'', include(tf_twilio_urls)),
+    re_path(r'', include(tf_urls)),
+    re_path(r'', include(tf_twilio_urls)),
```
Comment (claudep) [inline tests/urls.py:50]: Same remark than for main `urls.py`, `path` should be privileged.

File: two_factor/gateways/twilio/urls.py [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+from django.urls import re_path
 
 from .views import TwilioCallApp
 
 urlpatterns = ([
-    url(
-        regex=r'^twilio/inbound/two_factor/(?P<token>\d+)/$',
-        view=TwilioCallApp.as_view(),
+    re_path(
+        r'^twilio/inbound/two_factor/(?P<token>\d+)/$',
```
Comment (claudep) [inline two_factor/gateways/twilio/urls.py:7]: Even here, we could use `path('twilio/inbound/two_factor/<int:token>/',…)`

File: example/urls.py (hunk 1 above)
Comment (moggers87) [reply example/urls.py:42]: Using `re_path` means we're swapping like for like (`url` just calls `re_path`). I think we should leave migrating to `path` to another PR.

Comment (claudep) [reply example/urls.py:42]: Sure, that makes sense.

Comment (claudep) [reply example/urls.py:42]: Done in #374.

**Pull request merged**